examples/trading/agent.py

```diff
@@ -58,17 +63,19 @@
     api_key=os.getenv("OPENAI_API_KEY"),
 )
 
+wallet_manager = WalletManager(key_type=KeyType.SOLANA, key_path=os.getenv("SOLANA_KEY_PATH"))
+
 # Prepare a Web3 specific toolkit, relevant for the trading agent
 tools = [
-    coingecko.FetchMarketDataPerCategoriesTool(),
-    coingecko.GetCoinMarketDataTool(),
-    coingecko.GetCoinHistoricalDataTool(),
-    dexscreener.GetTokenDataTool(),
-    solana_common.GetAdminWalletAddressTool(),
-    solana_common.GetUserBalanceTool(),
-    raydium_openbook.BuyTokenWithSolTool(),
-    raydium_openbook.SellTokenForSolTool(),
-    jupiter.SwapTokenTool(),
+    coingecko.FetchMarketDataPerCategoriesTool(wallet_manager=wallet_manager),
+    coingecko.GetCoinMarketDataTool(wallet_manager=wallet_manager),
+    coingecko.GetCoinHistoricalDataTool(wallet_manager=wallet_manager),
+    dexscreener.GetTokenDataTool(wallet_manager=wallet_manager),
+    solana_native.GetSOLBalanceTool(wallet_manager=wallet_manager),
+    spl_token.GetTokenBalanceTool(wallet_manager=wallet_manager),
+    raydium_openbook.BuyTokenWithSolTool(wallet_manager=wallet_manager),
```

Comment:
This separation of `raydium_openbook` vs `raydium_cpmm` doesn't seem good. Should come from top level `raydium` namespace.